controllers/auth.go

```diff
@@ -61,3 +61,90 @@ func Login(c *fiber.Ctx) error {
 		"token": firebaseResponse.IDToken,
 	})
 }
+
+
+func Signup(c *fiber.Ctx) error {
+	var payload struct {
```

Comment:
podemos usar o "github.com/go-playground/validator/v10" para validar os campos enviados pelo front. 

```suggestion
	type Payload struct {
	Name     string `json:"name" validate:"required,min=2,max=100"`
	Email    string `json:"email" validate:"required,email"`
	Password string `json:"password" validate:"required,min=8"` // Reforçando segurança mínima
}
```